File: solutions/java/indexes.java
import com.splunk.*; 
import java.util.*;
import java.io.*;

public class indexes {

    public static void main(String[] args) {

		try {
			if(args.length < 1) {
				System.out.println("syntax: indexes.py [list | show indexname | clean indexname | create indexname | delete indexname]");
				System.exit(0);
			}
			Command command = Command.splunk("info").parse(args);
	        Service service = Service.connect(command.opts);

	        if(args[0].equalsIgnoreCase("list")) {
				for (String indexName : service.getIndexes().keySet())
					System.out.println("  " + indexName);
			}
			else if(args[0].equalsIgnoreCase("show")) {
				Index index = service.getIndexes().get(args[1]);
				for (String key : index.keySet())
					  System.out.println("  " + key + " = " + index.get(key));
			}
			else if(args[0].equalsIgnoreCase("clean")) {
				Index index = service.getIndexes().get(args[1]);
				index.clean(120);
				System.out.println("  index " + index.getName() + " cleaned.");
			}
			else if(args[0].equalsIgnoreCase("create")) {
				Index index = service.getIndexes().create(args[1]);
				System.out.println("  index " + index.getName() + " created.");
			}
			else if(args[0].equalsIgnoreCase("delete")) {
				service.getIndexes().remove(args[1]);
				System.out.println("  " + args[1] + " deleted.");
			}
			else {
				throw new Exception("Unknown command \"" + args[0] + "\"");
			}
			
		} 
    	catch (Exception e)  {
    		if(e.getMessage() != null) {
    			System.err.println("Error message: " + e.getMessage());
    		} else {
    			System.err.println("Error type: " + e.getClass());
    		}
		}
	}
}

File: solutions/python/indexes.py

import splunklib.client as client
import utils
import sys

opts = utils.parse(sys.argv[1:], {}, ".splunkrc")
try: 
    if len(sys.argv) == 1:
        print "syntax: indexes.py [list | show indexname | clean indexname | delete indexname | create indexname]"
        sys.exit()
    service = client.connect(**opts.kwargs)

    if sys.argv[1] == "list":
        for index in service.indexes:
            print "  %s" % (index.name)
    elif sys.argv[1] == "show":
        index = service.indexes[sys.argv[2]]
        for key in sorted(index.state.keys()):
            print '  ' + key
            for sub in index[key]:
                print "    %s: %s" % (sub, index[key][sub]) 
    elif sys.argv[1] == "create":
        index = service.indexes.create(sys.argv[2])
        print "  created index %s" % index.name
    elif sys.argv[1] == "clean":
        index = service.indexes[sys.argv[2]]
        index.clean()
        print "  cleaned index %s" % index.name
    elif sys.argv[1] == "delete":
        index = service.indexes[sys.argv[2]]
        index.delete()
        print "  index %s deleted" % sys.argv[2]
    else:
        raise Exception("unknown command \"%s\"" % sys.argv[1])

except Exception, err:
    sys.stderr.write('Error: %s\n' % str(err))
